plugins/woocommerce-blocks/assets/js/blocks/product-filter/block-variations.tsx

```diff
@@ -0,0 +1,105 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { BlockVariation } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import { activeFiltersIcon } from './inner-blocks/active-filters/icon';
+import { attributeFilterIcon } from './inner-blocks/attribute-filter/icon';
+import { priceFilterIcon } from './inner-blocks/price-filter/icon';
+import { ratingFilterIcon } from './inner-blocks/rating-filter/icon';
+import { stockStatusFilterIcon } from './inner-blocks/stock-filter/icon';
+
+const variations: BlockVariation[] = [
+	{
+		name: 'product-filter-active',
+		title: __( 'Active (Experimental)', 'woocommerce' ),
+		description: __(
+			'Display the currently active filters.',
+			'woocommerce'
+		),
+		attributes: {
+			heading: __( 'Active filters', 'woocommerce' ),
+			filterType: 'active-filters',
+		},
+		icon: {
+			src: activeFiltersIcon,
+		},
+		isDefault: true,
+	},
+	{
+		name: 'product-filter-price',
+		title: __( 'Price (Experimental)', 'woocommerce' ),
+		description: __(
+			'Enable customers to filter the product collection by choosing a price range.',
+			'woocommerce'
+		),
+		attributes: {
+			filterType: 'price-filter',
+			heading: __( 'Filter by Price', 'woocommerce' ),
+		},
+		icon: {
+			src: priceFilterIcon,
+		},
+	},
+	{
+		name: 'product-filter-stock-status',
+		title: __( 'Status (Experimental)', 'woocommerce' ),
+		description: __(
+			'Enable customers to filter the product collection by stock status.',
+			'woocommerce'
+		),
+		attributes: {
+			filterType: 'stock-filter',
+			heading: __( 'Filter by Stock Status', 'woocommerce' ),
+		},
+		icon: {
+			src: stockStatusFilterIcon,
+		},
+	},
+	{
+		name: 'product-filter-attribute',
+		title: __( 'Attribute (Experimental)', 'woocommerce' ),
+		description: __(
+			'Enable customers to filter the product collection by selecting one or more attributes, such as color.',
+			'woocommerce'
+		),
+		attributes: {
+			filterType: 'attribute-filter',
+			heading: __( 'Filter by Attribute', 'woocommerce' ),
+			attributeId: 0,
+		},
+		icon: {
+			src: attributeFilterIcon,
+		},
+	},
+	{
+		name: 'product-filter-rating',
+		title: __( 'Rating (Experimental)', 'woocommerce' ),
+		description: __(
+			'Enable customers to filter the product collection by rating.',
+			'woocommerce'
+		),
+		attributes: {
+			filterType: 'rating-filter',
+			heading: __( 'Filter by Rating', 'woocommerce' ),
+		},
+		icon: {
+			src: ratingFilterIcon,
+		},
+	},
+];
+
+/**
+ * Add `isActive` function to all Product Filter block variations.
+ * `isActive` function is used to find a variation match from a created
+ *  Block by providing its attributes.
+ */
+variations.forEach( ( variation ) => {
+	variation.isActive = () => [ 'filterType' ];
```

Comment:
This should be `string[]` or `() => boolean`, not `() => string[]`.
```suggestion
	variation.isActive = [ 'filterType' ];
```

The type is actually incorrect, we should fire a pull to fix it.
```
Diagnostics:
1. Type 'string[]' is not assignable to type '(blockAttributes: BlockAttributes, variationAttributes: BlockAttributes) => boolean | string[]'.
     Type 'string[]' provides no match for the signature '(blockAttributes: BlockAttributes, variationAttributes: BlockAttributes): boolean | string[]'. [2322]
```

Reply:
Fixed in b664a7b74f0ab496820b2e0601221eb07eab982b. From the type definition this is an acceptable way, but I changed to be an array of strings.